python_modules/dagster/dagster/_core/execution/context/init.py

```diff
@@ -129,56 +130,23 @@ def __init__(
         resources: Optional[Union[Resources, Dict[str, Any]]],
         instance: Optional[DagsterInstance],
     ):
-        from dagster._core.execution.api import ephemeral_instance_if_missing
-        from dagster._core.execution.build_resources import (
-            build_resources,
-            wrap_resources_for_execution,
-        )
         from dagster._core.execution.context_creation_pipeline import initialize_console_manager
 
-        self._instance_provided = (
-            check.opt_inst_param(instance, "instance", DagsterInstance) is not None
-        )
-        # Construct ephemeral instance if missing
-        self._instance_cm = ephemeral_instance_if_missing(instance)
-        # Pylint can't infer that the ephemeral_instance context manager has an __enter__ method,
-        # so ignore lint error
-        instance = self._instance_cm.__enter__()  # pylint: disable=no-member
-
-        # Shouldn't actually ever have a resources object directly from this initialization
-
-        self._resource_defs = wrap_resources_for_execution(
-            check.opt_dict_param(resources, "resources")
-        )
-
-        self._resources_cm = build_resources(self._resource_defs, instance=instance)
-        resources = self._resources_cm.__enter__()  # pylint: disable=no-member
-        self._resources_contain_cm = isinstance(resources, IContainsGenerator)
-
-        self._cm_scope_entered = False
-        super(UnboundInitResourceContext, self).__init__(
+        InitResourceContext.__init__(
+            self,
             resource_config=resource_config,
             resources=resources,
             resource_def=None,
             instance=instance,
             dagster_run=None,
             log_manager=initialize_console_manager(None),
         )
-
-    def __enter__(self):
-        self._cm_scope_entered = True
-        return self
-
-    def __exit__(self, *exc):
-        self._resources_cm.__exit__(*exc)  # pylint: disable=no-member
-        if self._instance_provided:
-            self._instance_cm.__exit__(*exc)  # pylint: disable=no-member
-
-    def __del__(self):
-        if self._resources_cm and self._resources_contain_cm and not self._cm_scope_entered:
-            self._resources_cm.__exit__(None, None, None)  # pylint: disable=no-member
-        if self._instance_provided and not self._cm_scope_entered:
-            self._instance_cm.__exit__(None, None, None)  # pylint: disable=no-member
+        UsesResourcesContext.__init__(
+            self,
+            resources=resources,
+            instance=instance,
+            context_str="init_resource",
+        )
```

Comment:
<!-- GRAPHITE_METADATA {"trueLocation":{"startLine":135,"endLine":149,"prLink":"https://app.graphite.dev/github/pr/dagster-io/dagster/9535"}} -->
<i class='graphite__hidden'>[Re: lines 135 to 149]<br/></i>

[1]<span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/9535?utm_source=unchanged-line-comment">Graphite</a>.</span>